src/app/pages/system/advanced/token-settings/token-settings.component.spec.ts

```diff
@@ -0,0 +1,67 @@
+import { HarnessLoader } from '@angular/cdk/testing';
+import { TestbedHarnessEnvironment } from '@angular/cdk/testing/testbed';
+import { ReactiveFormsModule } from '@angular/forms';
+import { MatButtonHarness } from '@angular/material/button/testing';
+import { createComponentFactory, mockProvider, Spectator } from '@ngneat/spectator/jest';
+import { provideMockStore } from '@ngrx/store/testing';
+import { mockWindow } from 'app/core/testing/utils/mock-window.utils';
+import { Preferences } from 'app/interfaces/preferences.interface';
+import { IxFormsModule } from 'app/modules/ix-forms/ix-forms.module';
+import { IxFormHarness } from 'app/modules/ix-forms/testing/ix-form.harness';
+import { TokenSettingsComponent } from 'app/pages/system/advanced/token-settings/token-settings.component';
+import { SystemGeneralService } from 'app/services';
+import { IxSlideInService } from 'app/services/ix-slide-in.service';
+import { selectPreferences } from 'app/store/preferences/preferences.selectors';
+
+describe('TokenSettingsComponent', () => {
+  let spectator: Spectator<TokenSettingsComponent>;
+  let loader: HarnessLoader;
+  const createComponent = createComponentFactory({
+    component: TokenSettingsComponent,
+    imports: [
+      IxFormsModule,
+      ReactiveFormsModule,
+    ],
+    providers: [
+      mockWindow({
+        localStorage: {
+          setItem: jest.fn,
+        },
+      }),
+      mockProvider(IxSlideInService),
+      mockProvider(SystemGeneralService),
+      provideMockStore({
+        selectors: [{
+          selector: selectPreferences,
+          value: { lifetime: 300 } as Preferences,
+        }],
+      }),
+    ],
+  });
+
+  beforeEach(() => {
+    spectator = createComponent();
+    loader = TestbedHarnessEnvironment.loader(spectator.fixture);
+  });
+
+  it('shows current lifetime values when form is being edited', async () => {
+    const form = await loader.getHarness(IxFormHarness);
+    const values = await form.getValues();
+
+    expect(values).toEqual({
+      'Token Lifetime': '300',
+    });
+  });
+
+  it('updates lifetime when save is pressed', async () => {
+    const form = await loader.getHarness(IxFormHarness);
+    await form.fillForm({
+      'Token Lifetime': '60',
+    });
+
+    const saveButton = await loader.getHarness(MatButtonHarness.with({ text: 'Save' }));
+    await saveButton.click();
```

Comment:
Need to actually check that something is being saved. You can spy on `store.dispatch`.